src/policy.c

```diff
@@ -863,23 +949,259 @@ pool_buildflavorcmp(Pool *pool, Solvable *s1, Solvable *s2)
     return 0;
   return pool_evrcmp_str(pool, f1 ? f1 : "" , f2 ? f2 : "", EVRCMP_COMPARE);
 }
-#endif
+
+void intersect_selection(Pool* pool, Id dep, Queue* prev)
+{
+  Queue tmp, res;
+  queue_init(&tmp);
+  // could use queue init buffer as optimization
+  queue_init(&res);
+
+  Id* pp, p;
+  pp = pool_whatprovides_ptr(pool, dep);
+  while ((p = *pp++) != 0)
+  {
+    queue_push(&tmp, p);
+  }
+
+  int i = 0, j = 0;
+  // set intersection, assuming sorted arrays
+  if (prev)
+  {
+    while (i < prev->count && j < tmp.count) {
+        if (prev->elements[i] < tmp.elements[j])
+            i++;
+        else if (tmp.elements[j] < prev->elements[i])
+            j++;
+        else /* if arr1[i] == arr2[j] */
+        {
+          queue_push(&res, tmp.elements[j]);
+          i++;
+          j++;
+        }
+    }
+  }
+  queue_free(&tmp);
+  queue_free(prev);
+  prev = &res;
+}
+
+int check_deps_unequal(Pool* pool, Queue* q1, Queue* q2, Id name)
+{
+  Id h1 = 0, h2 = 0;
+  int i;
+  Id elem;
+
+  // TODO check if reldep ID is equal if same reldep
+  Reldep* rdep;
+  for (i = 0; i < q1->count; ++i)
+  {
+    elem = q1->elements[i];
+    if (!ISRELDEP(elem)) continue;
+    rdep = GETRELDEP(pool, elem);
+
+    if (rdep->name == name)
+    {
+      h1 ^= rdep->evr;
```

Comment:
I took inspiration here from https://github.com/openSUSE/libsolv/blob/636c89b3888630ccb4f97bee3192dc6391c668eb/src/solvable.c#L522-L528

I hope this works

Reply:
turns out that this code didn't work for all cases :)